<launch>
  <arg name="robot_ws" default="~/diffy_ws"/>
  <machine name="robot" address="ip" env-loader="$(arg robot_ws)/src/diffy/envloader.sh" default="never" user="user" password="password"/>
  <machine name="station" address="localhost" env-loader="~/.bashrc" default="true"/>
  
  <param name="/robotlab/world" type="yaml" textfile="$(find robotlab)/launch/world_params.yaml"/>

  <group ns="measurement">
    <node machine="robot" pkg="diffy" type="recorder.py" name="recorder" respawn="false" required="true" output="log" args="-o rgb_images.tar.gz -t 1"/>
  </group>

  <!-- <group ns="robot_control">
    <node pkg="robotlab" type="controller.py" name="controller" required="true" output="screen"/>
    <node pkg="rospowerboard" type="teleop-client.py" name="client" required="true" output="log" args="1"/>
    <node machine="robot" pkg="rospowerboard" type="powerboard.py" name="motors" respawn="true" output="log" args="110"/>
  </group> -->

  <group ns="ground_truth">
    <!-- <node pkg="robotlab" type="camera.py" name="camera" respawn="false" required="true" output="log" args="-d"/> -->
    <!-- <node pkg="nodelet" type="nodelet" name="realsense2_camera_manager" required="false" respawn="false" output="log" args="manager"/> -->
    <!-- <node pkg="nodelet" type="nodelet" name="realsense2_camera" required="false" respawn="false" args="load realsense2_camera/RealSenseNodeFactory realsense2_camera_manager">
      <param name="serial_no"                type="str"  value=""/>
      <param name="usb_port_id"              type="str"  value=""/>
      <param name="device_type"              type="str"  value=""/>
      <param name="json_file_path"           type="str"  value=""/>
      <param name="rosbag_filename"          type="str"  value=""/>

      <param name="enable_pointcloud"        type="bool" value="false"/>
      <param name="pointcloud_texture_stream" type="str" value="RS2_STREAM_COLOR"/>
      <param name="pointcloud_texture_index"  type="int" value="0"/>
      <param name="allow_no_texture_points"  type="bool"   value="false"/>
      <param name="ordered_pc"               type="bool"   value="false"/>

      <param name="enable_sync"              type="bool" value="false"/>
      <param name="align_depth"              type="bool" value="false"/>

      <param name="fisheye_width"            type="int"  value="-1"/>
      <param name="fisheye_height"           type="int"  value="-1"/>
      <param name="enable_fisheye"           type="bool" value="false"/>
      <param name="enable_fisheye1"          type="bool" value="false"/>
      <param name="enable_fisheye2"          type="bool" value="false"/>

      <param name="depth_width"              type="int"  value="-1"/>
      <param name="depth_height"             type="int"  value="-1"/>
      <param name="enable_depth"             type="bool" value="false"/>

      <param name="confidence_width"         type="int"  value="-1"/>
      <param name="confidence_height"        type="int"  value="-1"/>
      <param name="enable_confidence"        type="bool" value="false"/>
      <param name="confidence_fps"           type="int"  value="-1"/>

      <param name="enable_color"             type="bool" value="true"/>
      <param name="color_width"              type="int"  value="640"/>
      <param name="color_height"             type="int"  value="480"/>
      <param name="color_fps"                type="int"  value="30"/>

      <param name="infra_width"              type="int"  value="848"/>
      <param name="infra_height"             type="int"  value="480"/>
      <param name="enable_infra"             type="bool" value="false"/>
      <param name="enable_infra1"            type="bool" value="false"/>
      <param name="enable_infra2"            type="bool" value="false"/>
      <param name="infra_rgb"                type="bool" value="false"/>

      <param name="fisheye_fps"              type="int"  value="-1"/>
      <param name="depth_fps"                type="int"  value="-1"/>
      <param name="infra_fps"                type="int"  value="-1"/>
      <param name="gyro_fps"                 type="int"  value="-1"/>
      <param name="accel_fps"                type="int"  value="-1"/>
      <param name="enable_gyro"              type="bool" value="false"/>
      <param name="enable_accel"             type="bool" value="false"/>
      <param name="enable_pose"              type="bool" value="false"/>

      <param name="base_frame_id"            type="str"  value="camera_link"/>
      <param name="depth_frame_id"           type="str"  value="camera_depth_frame"/>
      <param name="infra1_frame_id"          type="str"  value="camera_infra1_frame"/>
      <param name="infra2_frame_id"          type="str"  value="camera_infra2_frame"/>
      <param name="color_frame_id"           type="str"  value="camera_color_frame"/>
      <param name="fisheye_frame_id"         type="str"  value="camera_fisheye_frame"/>
      <param name="fisheye1_frame_id"        type="str"  value="camera_fisheye1_frame"/>
      <param name="fisheye2_frame_id"        type="str"  value="camera_fisheye2_frame"/>
      <param name="accel_frame_id"           type="str"  value="camera_accel_frame"/>
      <param name="gyro_frame_id"            type="str"  value="camera_gyro_frame"/>
      <param name="pose_frame_id"            type="str"  value="camera_pose_frame"/>

      <param name="depth_optical_frame_id"    type="str"  value="camera_depth_optical_frame"/>
      <param name="infra1_optical_frame_id"   type="str"  value="camera_infra1_optical_frame"/>
      <param name="infra2_optical_frame_id"   type="str"  value="camera_infra2_optical_frame"/>
      <param name="color_optical_frame_id"    type="str"  value="camera_color_optical_frame"/>
      <param name="fisheye_optical_frame_id"  type="str"  value="camera_fisheye_optical_frame"/>
      <param name="fisheye1_optical_frame_id" type="str"  value="camera_fisheye1_optical_frame"/>
      <param name="fisheye2_optical_frame_id" type="str"  value="camera_fisheye2_optical_frame"/>
      <param name="accel_optical_frame_id"    type="str"  value="camera_accel_optical_frame"/>
      <param name="gyro_optical_frame_id"     type="str"  value="camera_gyro_optical_frame"/>
      <param name="imu_optical_frame_id"      type="str"  value="camera_imu_optical_frame"/>
      <param name="pose_optical_frame_id"     type="str"  value="camera_pose_optical_frame"/>

      <param name="aligned_depth_to_color_frame_id"    type="str"  value="camera_aligned_depth_to_color_frame"/>
      <param name="aligned_depth_to_infra1_frame_id"   type="str"  value="camera_aligned_depth_to_infra1_frame"/>
      <param name="aligned_depth_to_infra2_frame_id"   type="str"  value="camera_aligned_depth_to_infra2_frame"/>
      <param name="aligned_depth_to_fisheye_frame_id"  type="str"  value="camera_aligned_depth_to_fisheye_frame"/>
      <param name="aligned_depth_to_fisheye1_frame_id" type="str"  value="camera_aligned_depth_to_fisheye1_frame"/>
      <param name="aligned_depth_to_fisheye2_frame_id" type="str"  value="camera_aligned_depth_to_fisheye2_frame"/>

      <param name="publish_tf"               type="bool"   value="true"/>
      <param name="tf_publish_rate"          type="double" value="0"/>

      <param name="odom_frame_id"            type="str"  value="camera_odom_frame"/>
      <param name="topic_odom_in"            type="str"  value="odom_in"/>
      <param name="calib_odom_file"          type="str"    value=""/>
      <param name="publish_odom_tf"          type="bool" value="true"/>
      <param name="stereo_module/exposure/1" type="int"  value="7500"/>
      <param name="stereo_module/gain/1"     type="int"  value="16"/>
      <param name="stereo_module/exposure/2" type="int"  value="1"/>
      <param name="stereo_module/gain/2"     type="int"  value="16"/>

      <param name="filters"                  type="str"    value=""/>
      <param name="clip_distance"            type="double" value="-2"/>
      <param name="linear_accel_cov"         type="double" value="0.01"/>
      <param name="initial_reset"            type="bool"   value="false"/>
      <param name="reconnect_timeout"        type="double" value="6.0"/>
      <param name="wait_for_device_timeout"  type="double" value="-1.0"/>
      <param name="unite_imu_method"         type="str"    value=""/>
    </node> -->
    <!-- <node pkg="robotlab" type="detector.py" name="detector" respawn="false" required="true" output="screen" args="-d"/> -->
  </group>

  <group ns="visuals">
    <!-- <node pkg="robotlab" type="display.py" name="display" respawn="false" output="log" required="true"/> -->
    <!-- <node pkg="robotlab" type="start.sh" name="server_starter" respawn="false" required="true" output="screen" args="$(find robotlab)/scripts"/> -->
  </group>
</launch>
